app/eth2wrap/eth2wrap.go

```diff
@@ -114,17 +114,17 @@ type multi struct {
 	selector *bestSelector
 }
 
-// bestIdx increments the selector with the best client index.
-func (m multi) bestIdx(i int) {
-	m.selector.Increment(i)
-}
-
 func (multi) Name() string {
 	return "eth2wrap.multi"
 }
 
 func (m multi) Address() string {
-	return m.clients[m.selector.Best()].Address()
+	address, ok := m.selector.BestAddress()
```

Comment:
In what case could `BestAddress` return `ok == false`?

Reply:
In theory this should never happen, because even after `period` elapsed, we always add at least one entry to the map. So, this is a safeguard in case somebody changes the logic in `Increment` and not look into this function.